code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -1,61 +1,187 @@
+	if(value == null)
+		if(reagents)
+			value = max(0, ceil((nutriments_value()*2 + water_value(FALSE)*0.5)/30 + satisfaction/2  - 0.5))
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/AVim()  
+//Appraise volume in milliliters. This is a very empirical approximation, but we have what we have at this moment
+//nutriments have "magical" volume, that not are visible at first look, watering reagents too have this "magic"
+	return reagents.total_volume + nutriments_value()/30*2 + water_value()/20*2
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/recalculate_bitesize()
+//Calculating biteamount if not set and 
+	if(reagents)
+		var/appraise_volume_in_milliliters = AVim()
+		if(appraise_volume_in_milliliters<100)
+			w_class = 1
 		else
-			bitesize = 2
-	value = 2*nutriment_amt
+			w_class = 2
+		if(!biteamount)
+			if(bitesize)
+				biteamount = ceil(reagents.total_volume/bitesize)
+			else	
+				//Let's take as the norm a volume equal to 35 milliliters per bite/drink
+				biteamount = ceil(appraise_volume_in_milliliters/35)
+		if(!bitesize)
+			bitesize = ceil(reagents.total_volume*10/biteamount)/10
+
+/obj/item/weapon/reagent_containers/food/snacks/New() 
+	..()
+	if(nutriment_amt>0)
+		reagents.add_reagent("nutriment", nutriment_amt, nutriment_desc)
+	spawn(1) //because we need to work new() proc of child before this calculations
+		recalculate_bitesize()
+		appraise()
+		if(debug_snacks) //for debugging log of calculated values snacks creating on/off it by set TRUE/FALSE at begining of snacks.dm (this file)
+			message_admins("Created snack: [name]. Bite amount: [biteamount], bite size: [bitesize]. Value: [value].")
+			if(reagents)
+				message_admins("Size: [round((AVim()*100)/100)] milliliters")
+			else
+				message_admins("SIZE NOT DEFINED! NO REAGENTS!!!")
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/over(var/mob/M) //end of snack
+	usr.drop_from_inventory(src)	//so icons update :[
+	if(trash)
+		if(ispath(trash,/obj/item))
+			var/obj/item/TrashItem = new trash(usr)
+			usr.put_in_hands(TrashItem)
+		else if(istype(trash,/obj/item))
+			usr.put_in_hands(trash)
+	qdel(src)
 
-	//Placeholder for effect that trigger on eating that aren't tied to reagents.
+//Placeholder for effect that trigger on eating that aren't tied to reagents.
 /obj/item/weapon/reagent_containers/food/snacks/proc/On_Consume(var/mob/M)
-	if (!usr)	return
-	if (raw)
-		if (ishuman(M))
-			var/mob/living/human/H = M
-			if (!H.orc && !H.crab && !H.wolfman && !H.lizard)
-				M.reagents.add_reagent("food_poisoning", 1)
-	if (ishuman(M))
-		var/mob/living/human/HM = M
-		if (HM.orc || HM.crab || HM.wolfman)
-			HM.mood += abs(satisfaction)
+	if (!usr)
+		return
+	var/mob/living/human/H = M
+	if (ishuman(H))
+		if (H.orc || H.crab || H.wolfman)
+			H.mood += abs(satisfaction*bitesize) //orcs, crabs and wolfmans will be get satisfaction positive anyway from any food
```

Comment:
need to correct that to: += abs(ceil(satisfaction*bitesize/reagents.total_volume)